### DISCUSSION TITLE
微信公众号网页授权 snsapi_base 获取不到openid

### DISCUSSION THREAD

**Discussion by xshwy** [w7corp/easywechat#2623] [Q&A]
```php 
// Laravel, easywechat 6.9.3
$app = new Application($config);
        $oauth = $app->getOauth();
        $user = $oauth->userFromCode($code);
        var_dump($user->toJson());
```
输出的结果是
```
{
  "id": null,
  "name": null,
  "nickname": null,
  "avatar": null,
  "email": null,
  "raw": {
    "errcode": 48001,
    "errmsg": "api unauthorized, rid: 637799ac-32820464-029b2916"
  },
  "access_token": "62_YPg4Q……RB2EFP0",
  "refresh_token": "62_TvV……-yQnDTw",
  "expires_in": 7200,
  "token_response": {
    "access_token": "62_YPg……B2EFP0",
    "expires_in": 7200,
    "refresh_token": "62_Tv……y-yQnDTw",
    "openid": "oEJ……DThZM",
    "scope": "snsapi_base"
  }
}
```
通过`$user->getId();` 获取的结果为NULL，但是在`token_response`中是有openid的，请问是我的使用姿势不对，还是需要再次进行什么操作？

**Comment by xshwy** [comment]
`$user['token_response']['openid']` 是可以正常获取到openid的

**Reply by overtrue** [reply]
`$config` 打印一下

**Reply by xshwy** [reply]
```
     $config = [
          'app_id' => 'wx61……748f', // 公众号appid
          'secret' => '40f75……9ee36',
          'aes_key' => '' // 明文模式请勿填写 EncodingAESKey
      ];
```

感谢回复，我公众号的配置如上，因为只需要读取一个openid，所以没有填写其他信息。

**Reply by iluyuns** [reply]
$oauth->userFromCode($code);  调用这个前，请先配置scope  如  `$oauth->scopes(['snsapi_base'])->userFromCode($code);`

**Comment by shp1102** [comment]
还没解决吗？我也遇到这个问题了

**Reply by xshwy** [reply]
目前我使用 `$user['token_response']['openid'] ` 是可以正常获取到openid的，你试试看。

**Reply by iluyuns** [reply]
$oauth->userFromCode($code); 调用这个前，请先配置scope 如 $oauth->scopes(['snsapi_base'])->userFromCode($code);

**Comment by hankin-h-k** [comment]
出现同样的问题, 偶尔获取不到openid，求教什么配置有问题吗？
返回数据 ：{
  "id": null,
  "name": null,
  "nickname": null,
  "avatar": null,
  "email": null,
}
easywechat：6.7
配置信息：
![Snipaste_2023-10-11_15-06-17](https://github.com/w7corp/easywechat/assets/24538175/0d8590b1-ad2e-4bca-9fbc-d0c6b1db14fb)

**Reply by unsetMe** [reply]
有可能是ip白名单的问题

**Comment by iluyuns** [comment]
$oauth->userFromCode($code); 调用这个前，请先配置scope 如 $oauth->scopes(['snsapi_base'])->userFromCode($code);
